### ISSUE TITLE
rom0 und 1 remaping

### ISSUE THREAD

**Issue by grafzahl12345** [mattcallow/CPC_PICOROM#6]
Hello,
is there a way to use picorom without remaping rom0 and rom1. On my stock cfc 6218 both rom's are not remapable without reworking cs pins of both rom's. So is there a software option in picorom firmware ?
thx

**Comment by mattcallow** [comment]
could you please explain further? My understanding is that the ROMDIS pin will disable the basic ROM and firmware ROM. on the 6128. But, there is an issue with the DOS ROM in the 6128. See #7

**Comment by grafzahl12345** [comment]
hello and thank you for your replay. 
Sorry that i'm so late with my answer. Today i updated my rev.1 pcb with the 2,2k resistor and  the firmware. At first it didn't work, so i proped  /OE pin of both bios roms and they where high. Startup doen't work, i only got a bootloop with blue screen and the Amstrad logo. After i removed both roms out of thier sockets bootup was a sucess. But the system wasn't stable only useabel with original rom7. Unirom, parados1.2 got the pico to crash. Maxam 1.5,pacman.rom und AmDiagU are booting. Amstrad Diag crashed by testing roms up to slot 4.
|PUSB, |LED and |ROMSET commands work, the other commands crash the pico. You have to use  Pico Reset button to get the system to start. If you use cpc restet you got the last screen with bury fonts from bevor the crash. 
Is there a pico firmware with bootlog ? 
thx for your work

**Comment by mattcallow** [comment]
Are you using the latest code from the main branch? It was updated yesterday to fix some problems with the 6128
You will not be able to use ROM7 in the pico, since the DOS ROM in the 6128 cannot be disabled.
I'm planning to release another firmware version which will include some debug logs.

**Comment by grafzahl12345** [comment]
yes i tried it today and i had to insert my Amdrad 40015 rom7 eprom for using firmware of the tree  issue7. For default branch firmware i had to  remove both roms. My amstrad doesn't work with picorom if the eproms are pressent, that's why i asked if there is a option to disable rom0 & rom1 in picorom. 

![Image](https://github.com/user-attachments/assets/671472d3-6ab5-4202-bbbd-45a76f2a1bb9)

**Comment by mattcallow** [comment]
First, make sure that you have the latest firmware and ROM image from the main branch. I can't tell the position of the two switches from the picture, but I assume you have them set to OFF ON. 
What do you mean by ROM0 and ROM1? Do you mean the two ROMs on the Amstrad main board? If so, then IC103, is disabled by the right hand switch on the picoROM, which connects ROMDIS to +5V
The DOS ROM, IC204 cannot be disabled by the picoROM, so you must make sure that you don't have a ROM7 in your config.
Can you post your config file?

**Comment by grafzahl12345** [comment]
so i got it all working ;-). I had to compile it myself. The error may lay in your prompt to compile picorom.rom by cmake. It allways got me errors. After using console and comiling it by commandline the  picorom.rom worked with all command on my cfc6218. The only thing is for using rom7 you have to remove it physicaly. I add snapshoots from my boots and config files. 
thx for your support 

[debug.txt](https://github.com/user-attachments/files/18628887/debug.txt)

my default.cfg

# Default config
L:OS_6128.ROM
0:BASIC_1.1.ROM
1:PICOROM.ROM
2:AmDiagU.rom
3:maxam15.rom
4:FOSP-A.ROM
5:FOSP-B.ROM
6:FOSP-C.ROM
7:AMSDOS_0.7.ROM
8:FOSP-D.ROM
9:CATRUN12.ROM
10:HXC35.ROM

my unidos.cfg

#unidos config
L:OS_6128.ROM
0:BASIC_1.1.ROM
1:PICOROM.ROM
2:nova.rom
3:maxam15.rom
4:cpm_ac.rom
5:cpm1.rom
6:cpm2.rom
7:unidos.rom
8:HXC25.ROM
9:unitools.rom
10:parados12.rom
11:albireo.rom
12:AmDiagU.rom
13:CATRUN12.ROM

![Image](https://github.com/user-attachments/assets/2ac9c125-4b5e-47c7-86b1-46a63b3b71f5)

![Image](https://github.com/user-attachments/assets/3da84600-233d-442f-8d8a-0daf8fcd9255)

![Image](https://github.com/user-attachments/assets/651c3072-f18b-4393-8da3-e55d17f88b5a)

![Image](https://github.com/user-attachments/assets/006d5750-dc29-4ca5-b1a0-ad32353d2ad8)
